lib/std/c.zig

```diff
@@ -124,7 +124,14 @@ pub extern "c" fn readlinkat(dirfd: fd_t, noalias path: [*:0]const u8, noalias b
 pub usingnamespace switch (builtin.os.tag) {
     .macos, .ios, .watchos, .tvos => struct {
         pub const realpath = @"realpath$DARWIN_EXTSN";
-        pub const fstatat = @"fstatat$INODE64";
+        pub usingnamespace switch (builtin.arch) {
+            .aarch64 => struct {
+                pub extern "c" fn fstatat(dirfd: fd_t, path: [*:0]const u8, stat_buf: *Stat, flags: u32) c_int;
+            },
+            else => struct {
+                pub const fstatat = @"fstatat$INODE64";
+            },
+        };
```

Comment:
Would it maybe be cleaner to simply list both functions and rely on lazy evaluation?

```zig
pub const fstatat = switch (builtin.arch) {
    .aarch64 => fstatat,
    else => @"fstatat$INODE64",
};
```

Reply:
Excellent suggestion!

Reply:
Hmm, any ideas if there's a way to avoid overriding the previous definition, i.e., `fstatat` has already been defined by `pub extern "c" fn fstatat...`?

Reply:
Take a step back and reorganize things. Is the previous definition correct already? Can just use that. If not, sounds like fstatat needs to become one of the functions that moves from the common file to OS-specific files.

Reply:
Let's try to keep _all_ the os-related conditional definitions/renames in the switch below.

> Hmm, any ideas if there's a way to avoid overriding the previous definition, i.e., fstatat has already been defined by pub extern "c" fn fstatat...?

You can handle it the same way `fstat` is handled below: define `fstatat` in `c/darwin.zig` together with its weird cousin `fstatat$INODE64` (and add a comment explaining why it's defined there) and then export the correct `fstat` symbol below. Other platforms get the usual `pub extern "c" fn fstatat(..) ...`, it's a bit repetitive but I couldn't figure out a nicer looking way to handle this symbol mess.

Reply:
Addressed in 4f50958. Have a look and lemme know if that's what you had in mind.

Reply:
Looks good.
Perhaps it's time to think of moving all the c definitions in a separate file for each os.

Reply:
Sounds like a good idea!

Reply:
Orrr... we could resurrect Andrew's idea of an `@extern` suggested in #3971 and augment it with a way to specify a comptime string as symbol name. This way we finally get the weak-symbol mechanism and gain a cleaner way to deal with those nasty renames.

Reply:
Hmm, after a quick browse I think I'm partial to the second solution of weak-symbol mechanism. This would indeed make a lot of things cleaner and more elegant.